e2e/pod.go

```diff
@@ -26,38 +28,32 @@ func getDebugPodName(ctx context.Context, kube *Kubeclient) (string, error) {
 	return podName, nil
 }
 
-func getPodIP(ctx context.Context, kube *Kubeclient, namespaceName, podName string) (string, error) {
-	pod, err := kube.Typed.CoreV1().Pods(namespaceName).Get(ctx, podName, metav1.GetOptions{})
-	if err != nil {
-		return "", fmt.Errorf("unable to get pod %s/%s: %w", namespaceName, podName, err)
-	}
-	return pod.Status.PodIP, nil
-}
-
-func ensureTestNginxPod(ctx context.Context, kube *Kubeclient, nodeName string) (string, error) {
+func ensureTestNginxPod(ctx context.Context, t *testing.T, namespace string, kube *Kubeclient, nodeName string) (string, error) {
 	nginxPodName := fmt.Sprintf("%s-nginx", nodeName)
 	nginxPodManifest := getNginxPodTemplate(nodeName)
-	if err := ensurePod(ctx, kube, nginxPodName, nginxPodManifest); err != nil {
+	if err := ensurePod(ctx, t, namespace, kube, nginxPodName, nginxPodManifest); err != nil {
 		return "", fmt.Errorf("failed to ensure test nginx pod %q: %w", nginxPodName, err)
 	}
 	return nginxPodName, nil
 }
 
-func ensureWasmPods(ctx context.Context, kube *Kubeclient, nodeName string) (string, error) {
+func ensureWasmPod(ctx context.Context, t *testing.T, namespace string, kube *Kubeclient, nodeName string) (string, error) {
```

Comment:
do we really need this `ensureWasmPod` function anymore? I'd probably opt for inlining this stuff directly in the validator implementation

Reply:
inlined